### DISCUSSION TITLE
Upgrade attempts Failing need help.

### DISCUSSION THREAD

**Discussion by mcooperRPI** [netbox-community/netbox-docker#991] [General]
Hello Guys,

I have inherited a netbox 2.6.2 production box from before I got here. Now they want me to upgrade it and I have tried for the past month everything I have tried has failed. I am asking for help to get this done. My latest failure came from this [https://www.avecseclabs.com/bif/2020/05/15/upgrading-docker-based-netbox-ipam-while-preserving-stored-data/](url) The error follows:

_[root@vs-dub-netbox-01 netbox-docker]# docker-compose rm -fs netbox netbox-worker postgres
ERROR: The Compose file './docker-compose.yml' is invalid because:
services.netbox-housekeeping.depends_on contains an invalid type, it should be an array
services.netbox-worker.depends_on contains an invalid type, it should be an array_

Thanks guys,
CentOS 7 Server with Netbox 2.6.2 and Postgresql 9.6, I need to go to Netbox 3.4.8 and Postgresql 12.14 on Ubuntu Server 20.04

**Comment by kkthxbye-code** [comment]
The specific error is because your docker compose version is too old. Remember you cannot upgrade directly from pre 2.10 to 3.0+. You should read all release notes for all major and minor versions between your current version and your target version. You can skip patch version release notes.

As a reminder, versioning for netbox is major.minor.patch. it might be necessary to read the netbox-docker release notes as well.

**Comment by mcooperRPI** [comment]
I did unfortunately I don’t know what I am looking for. The release notes don’t make sense to me sorry. Thanks, Michael CooperCertified Linux ProfessionalKubernetes/DockerRPi Hobby Blog From: kkthxbyeSent: Wednesday, April 19, 2023 5:06 PMTo: netbox-community/netbox-dockerCc: Michael Cooper; AuthorSubject: Re: [netbox-community/netbox-docker] Upgrade attempts Failing need help. (Discussion #991) The specific error is because your docker compose version is too old. Remember you cannot upgrade directly from pre 2.10 to 3.0+. You should read all release notes for all major and minor versions between your current version and your target version. You can skip patch version release notes.As a reminder, versioning for netbox is major.minor.patch. it might be necessary to read the netbox-docker release notes as well.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you authored the thread.Message ID: ***@***.***>
